src/components/TasksTable/index.jsx

```diff
@@ -42,19 +49,21 @@ export default class TasksTable extends Component {
   };
 
   getTableData = memoizeWith(
-    (sortBy, sortDirection, items) => {
+    (sortBy, sortDirection, tag, items) => {
       const ids = sorted(items);
 
-      return `${ids.join('-')}-${sortBy}-${sortDirection}`;
+      return `${ids.join('-')}-${sortBy}-${sortDirection}-${tag}`;
     },
-    (sortBy, sortDirection, items) => {
+    (sortBy, sortDirection, tag, items) => {
       const sortByProperty = camelCase(sortBy);
 
       if (!sortBy) {
         return items;
       }
 
-      return [...items].sort((a, b) => {
+      return [
+        ...items.filter(item => !tag || item.tags.indexOf(tag) > -1),
+      ].sort((a, b) => {
```

Comment:
The filter() method creates a new array so we don't need the `[...items]` trick anymore. Also, use `.includes()` instead of `indexOf()` when you can.

```jsx
items
  .filter(item => !tag || item.tags.includes(tag))
  .sort(...)
```